prog/rand.go

```diff
@@ -770,9 +774,16 @@ func (r *randGen) generateArg(s *state, typ sys.Type, dir ArgDir, sizes map[stri
 		filename := r.filename(s)
 		return dataArg([]byte(filename)), nil, nil
 	case sys.ArrayType:
-		count := a.Len
-		if count == 0 {
+		count := uintptr(0)
+		switch a.Kind {
+		case sys.ArrayRandLen:
 			count = r.rand(6)
+			a.Len = count
```

Comment:
Don't assign to a.Len.
'a' is meta information associated with a syscall in general, not with a particular invocation of that syscall in a particular program.
Moreover, here 'a' is a local copy of that meta information.